src/main/java/org/folio/dew/batch/marc/MarcWriter.java

```diff
@@ -1,40 +1,16 @@
 package org.folio.dew.batch.marc;
 
 import lombok.extern.log4j.Log4j2;
-import org.marc4j.MarcJsonWriter;
+import org.folio.dew.batch.AbstractStorageStreamWriter;
+import org.folio.dew.repository.LocalFilesStorage;
+import org.folio.dew.repository.S3CompatibleResource;
 import org.marc4j.marc.Record;
-import org.springframework.batch.item.file.FlatFileItemWriter;
-import org.springframework.core.io.FileSystemResource;
-
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.util.List;
 
 @Log4j2
-public class MarcWriter extends FlatFileItemWriter<Record> {
-
-  public MarcWriter(String tempOutputFilePath) {
-    setAppendAllowed(true);
-    setResource(new FileSystemResource(tempOutputFilePath));
+public class MarcWriter extends AbstractStorageStreamWriter<Record, LocalFilesStorage> {
+  public MarcWriter(String tempOutputFilePath, LocalFilesStorage localFilesStorage) {
+    super(tempOutputFilePath, localFilesStorage);
```

Comment:
Basically super invokes parent constructor that does not set any fields. As option we can impement this empty constructor and setResource() call will not be needed